### PR TITLE
bring the `ffi` module to 3.2.x

### PR DESCRIPTION
**Pull request by cdisselkoen**
## Description of changes

`main` has a bunch of great FFI changes that are scheduled for release in 4.0, but some users would like a 3.x-compatible version of `cedar-wasm`.  Obviously, changes to the structs in the `cedar_policy::frontend` module are breaking, so we can’t backport them to 3.x.  But, since we are planning to rename `frontend` to `ffi` in 4.0 anyway, this PR proposes release the new FFI interface in the `ffi` module on 3.x, so that 3.2+ will have both the `frontend` and `ffi` modules and consumers can use either.  Existing users of `frontend` are not broken as there are no changes in  the `frontend` module, and since `cedar-wasm` has not been released yet, there’s no harm in having it be based on the `ffi` interface even on 3.x.

## Checklist for requesting a review

The change in this PR is (choose one, and delete the other options):

- [x] A backwards-compatible change requiring a minor version bump to `cedar-policy` (e.g., addition of a new API).

I confirm that this PR (choose one, and delete the other options):

- [x] Updates the "Unreleased" section of the CHANGELOG with a description of my change (required for major/minor version bumps).

I confirm that [`cedar-spec`](https://github.com/cedar-policy/cedar-spec) (choose one, and delete the other options):

- [x] Does not require updates because my change does not impact the Cedar formal model or DRT infrastructure.

### CODE REVIEW

File: cedar-testing/src/cedar_test_impl.rs [hunk 1]
```diff
@@ -19,7 +19,8 @@
 //! running the integration tests and for performing randomized differential
 //! testing (see <https://github.com/cedar-policy/cedar-spec>).
 
-pub use cedar_policy::frontend::is_authorized::InterfaceResponse;
+#![allow(deprecated)] // Functionality in this module depends on `frontend`. probably covered by https://github.com/cedar-policy/cedar/issues/821 ?
```
Comment (khieta) [inline cedar-testing/src/cedar_test_impl.rs:22]: Yes, this will be addressed as part of #821

**Comment by cdisselkoen** [comment]
Apparently I don't have enough `#[allow(deprecated)]` already, for the `frontend`'s usage of its own types and functions

**Comment by cdisselkoen** [comment]
Also worth noting that this doesn't include the changes in #837 because #837 is not merged yet.  If this gets merged before #837, we'll backport #837 separately.  If #837 gets merged first, I'll adapt this PR.

File: cedar-policy/CHANGELOG.md [hunk 2]
```diff
@@ -423,7 +431,8 @@ Cedar Language Version: 2.0.0
 Cedar Language Version: 2.0.0
 - Initial release of `cedar-policy`.
 
-[Unreleased]: https://github.com/cedar-policy/cedar/compare/v3.1.3...main
+[Unreleased]: https://github.com/cedar-policy/cedar/compare/v3.2.0...main
+[3.2.0]: https://github.com/cedar-policy/cedar/compare/v3.1.3...v3.2.0
```
Comment (khieta) [inline cedar-policy/CHANGELOG.md:435]: Also done in #851

**Review by khieta — APPROVED** [review]
PR looks fine, but it occurs to me that we never actually moved any `cedar-wasm/` code back to the 3.2 release branch. Are you planning on doing that next?

File: cedar-policy/CHANGELOG.md (hunk 2 above)
Comment (cdisselkoen) [reply cedar-policy/CHANGELOG.md:435]: #851 is towards `main` and this is towards `release/3.2.x`.  Ideally we'd have #851's changes on `release/3.2.x` before we merge this, and I'll rebase.

**Comment by cdisselkoen** [comment]
Yes, I'm planning on bringing `cedar-wasm` in a separate PR

File: cedar-policy/CHANGELOG.md (hunk 2 above)
Comment (cdisselkoen) [reply cedar-policy/CHANGELOG.md:435]: #853 brings #851's changes to `release/3.2.x`

**Comment by cdisselkoen** [comment]
It seems these warnings are not silenceable; upstream Rust issue https://github.com/rust-lang/rust/issues/47238.  Since we deny warnings, we'll need a workaround here.

**Comment by john-h-kastner-aws** [comment]
FYI: Java CI failures are expected pending cedar-policy/cedar-java#141, but the cedar-drt CI should pass

**Review by shaobo-he-aws — APPROVED** [review]

**Review by Swolebrain — APPROVED** [review]

**Review by khieta — APPROVED** [review]

**Pull request merged**